mono/metadata/w32file.c

```diff
@@ -374,40 +374,46 @@ get_filesystem_entries (const gunichar2 *path,
 }
 
 
-MonoArray *
-ves_icall_System_IO_MonoIO_GetFileSystemEntries (MonoString *path,
-						 MonoString *path_with_pattern,
+MonoArrayHandle
+ves_icall_System_IO_MonoIO_GetFileSystemEntries (MonoStringHandle path,
+						 MonoStringHandle path_with_pattern,
 						 gint attrs, gint mask,
 						 gint32 *ioerror)
 {
 	MonoError error;
```

Comment:
As Ludovic mentioned, functions declared with `HANDLES()` take a final `MonoError *error` argument.  So you shouldn't declare one here.